src/components/input/test/Select-test.js

```diff
@@ -425,12 +436,137 @@ describe('Select', () => {
     });
   });
 
+  context('when listbox does not fit in available space', () => {
+    async function getOpenListbox(wrapper) {
+      toggleListbox(wrapper);
+      await waitFor(() => !isListboxClosed(wrapper));
+
+      return getListbox(wrapper);
+    }
+
+    it('never renders a listbox bigger than the viewport', async () => {
+      const name = 'name'.repeat(1000);
+      const wrapper = createComponent(
+        { buttonContent: 'Select a value' },
+        {
+          items: ['1', '2', '3'].map(id => ({ id, name })),
+        },
+      );
+
+      const listbox = await getOpenListbox(wrapper);
+      const { width: listboxWidth } = listbox
+        .getDOMNode()
+        .getBoundingClientRect();
+
+      assert.isTrue(listboxWidth < window.window.innerWidth);
+    });
+
+    [
+      { name: 'long name'.repeat(50), shouldGrow: true },
+      { name: 'short name', shouldGrow: false },
+    ].forEach(({ name, shouldGrow }) => {
+      it('makes left-aligned Select listbox grow to the left if there is space', async () => {
+        const wrapper = createComponent(
+          { buttonContent: 'Select a value' },
+          {
+            items: ['1', '2', '3'].map(id => ({ id, name })),
+          },
+        );
+
+        const listbox = await getOpenListbox(wrapper);
+        const listboxStyleLeft = listbox.getDOMNode().style.left;
+        const listboxLeft = Number(listboxStyleLeft.replace('px', ''));
```

Comment:
This is fine. You can use `parseInt` on a string like "42px" or `parseFloat` if the value may be a float, and get the expected result. This is because these functions stop parsing once they encounter a character that is not a valid digit.